feat(web): ✨ Add a user profile dropdown and overhaul navbar

Steveplays28 opened the pull request.

Steveplays28 committed:
feat(web): :sparkles: Add a user profile dropdown menu

Steveplays28 committed:
feat(web): :sparkles: Add "Open Reports" button to the navbar dropdown

Steveplays28 committed:
fix(web): :lipstick: Rename the "Open Reports" button to "Reports"

Steveplays28 committed:
feat(web): :sparkles: Add "Profile" and "Reports" buttons to the mobile navbar

Also added a profile icon placeholder next to the username.

Steveplays28 committed:
fix(web): :lipstick: Make the color of the log out icon red on mobile

rotgruengelb reviewed (changes requested):
Use the correct styling variables instead

Steveplays28 committed:
fix(web): :lipstick: Replace the log out button's hard-coded color with a button variant

rotgruengelb committed:
feat(web): new navbar

Steveplays28 committed:
fix(web): :bug: Fix hover effects and re-add "Reports" button

Also removed the "Dashboard" and "Moderation Dashboard" buttons from the profile dropdown (desktop).

Steveplays28 committed:
fix(web): :bug: Remove `item.desktopOnly` check
